pkg/util/metric/aggmetric/agg_metric.go

```diff
@@ -22,6 +22,18 @@ import (
 
 var delimiter = []byte{'_'}
 
+const (
+	dbLabel  = "database"
+	appLabel = "application_name"
+)
+
+const (
+	LabelConfigDisabled = iota
+	LabelConfigApp
+	LabelConfigDB
+	LabelConfigAppAndDB
+)
```

Comment:
Non-blocking NIT: 
I think this is a good candidate for using bitwise flags. It will allow us to extend this list in the future without making the evaluation of these flags complex. Options like `LabelConfigAppAndDB` will become complex if/when there are more labels.

```suggestion
const (
    LabelConfigDisabled = 0
    LabelConfigDB       = 1 << iota
    LabelConfigApp                  
)

// combination can be represented as
const bothLabels = LabelConfigApp | LabelConfigDB
```

And they can be evaluated like this:
```go
var childMetric ChildMetric
switch {
case c.labelConfig.Load()&LabelConfigAppAndDB == LabelConfigAppAndDB:
    childMetric = c.getOrAddChild(db, app)
case c.labelConfig.Load()&LabelConfigDB != 0:
    childMetric = c.getOrAddChild(db)
case c.labelConfig.Load()&LabelConfigApp != 0:
    childMetric = c.getOrAddChild(app)
default:
    return
}
```

Reply:
Let me try to evaluate and update as part of separate PR.
<!-- Reviewable comment -ONsPTdr6Z2a7uhAv3IV:b-llcr4 -->
<!-- Sent from Reviewable.io -->